doc/specs/stdlib_stats_distribution.md

```diff
@@ -0,0 +1,456 @@
+---
+title: stats_distribution
+---
+
+# Statistical Distribution
+
+[TOC]
+
+## `random_seed` - set or get a value of seed to the random distribution psudorandom number generator
+
+### Status
+
+Experimental
+
+### Description
+
+Must set the seed value before calling random distribution for variates. 
+
+### Syntax
+
+`call random_seed(put [, get])`
+
+### Arguments
+
+`put`: argument has intent `in` and may be a scalar of type `integer` with kind of int32.
+
+`get`: optional argument has intent `out` and is a scalar of type `integer` with kind of int32.
+
+### Return value
+
+
+### Example
+
+```fortran
+program demo_norm_seed
+    use stdlib_stats_distribution, only : random_seed
+    use iso_fortran_env, only : int32
+    implicit none
+    integer(int32) :: seed, seed_value
+
+    seed = 1234567
+    call random_seed(seed)                   ! reset the value of seed
+    call random_seed(seed, seed_value)       ! get current value of seed
+end program demo_norm_seed
+```
+
+## `uniform_distribution_rvs` - uniform distribution random variates
+
+### Status
+
+Experimental
+
+### Description
+
+Using the parameters `loc` and `scale`, one obtains the uniformly distributed random variates on [loc, loc + scale]. Return a scalar of type `real` pseudorandom number or a rank one array of such numbers with a size of `array_size`.
+
+### Syntax
+
+`result = [[stdlib_stats_distribution(module):uniform_distribution_rvs(interface)]](loc, scale, [array_size])`
+
+### Arguments
+
+`array_size`: optional argument has intent `in` and is a scalar of type `integer`.
+
+`loc`: has intent `in` and is a scalar of type `real`.
+
+`scale`: has intent `in` and is a scalar of type `real`.
+
+### Return value
+
+The result is a scalar of type `real`, or is a rank one array of type `real` with size of `array_size`.
+
+### Example
+
+```fortran
+program demo_uniform_rvs
+    use stdlib_stats_distribution, only: uniform => uniform_distribution_rvs
+    implicit none
+    real :: loc, scale, rv(10)
+
+    loc = 0.0
+    scale = 1.0
+    print *, uniform(loc, scale)            ! single standard uniform random variate in [0, 1)
+    rv = uniform(loc, scale, 10)            ! an array of 10 uniform random variates in [0, 1)
+    loc = -1.0
+    scale = 2.0
+    print *, uniform(loc, scale)            ! single uniform random variate in [-1,1)
+    loc = -0.5
+    scale = 1.0
+    rv = uniform(loc, scale, 10)            ! an array of 10 uniform variates in [-0.5, 0.5)
+end program demo_uniform_rvs
+```
+
+## `uniform_distribution_pdf` - uniform probability density function
+
+### Status
+
+Experimental
+
+### Description
+
+The probability density function of the continuous uniform distribution.
+
+![equation](https://latex.codecogs.com/gif.latex?f(x)=\begin{cases}\frac{1}{scale}&loc\leqslant&space;x<loc&plus;scale\\\\0&x<loc,or,x>loc&plus;scale\end{cases})
```

Comment:
FORD supports Latex: see [here](https://github.com/Fortran-FOSS-Programmers/ford/wiki/Writing-Documentation#latex-support) for more details.
Something like (I didn't check the formula):
```suggestion
\( \frac{1}{scale}&loc\leqslant&space;x<loc&plus;scale\\\\0&x<loc,or,x>loc&plus;scale )\
```

Reply:
> FORD supports Latex: see [here](https://github.com/Fortran-FOSS-Programmers/ford/wiki/Writing-Documentation#latex-support) for more details.
> Something like (I didn't check the formula):

I tried FORD LaTex support, It seems not working for my formulae.

Reply:
Ok. Thanks for trying. I will try it later too. IMO it is preferable to not rely on a third party app.